File: species/fixture/testdata/unused-variable/repo/repo.go
package unusedvariable

// scaleFactor is the package's real, referenced state — it has no marker and is
// read by Scale below, so the species leaves it alone (and the compile gate
// would skip its removal anyway).
var scaleFactor = 3

//ant:unused-variable scratch was a leftover constant nothing reads anymore
var unusedScratch = 41 + 1

// Scale is the package's real exported logic; it reads scaleFactor, proving the
// fix removes ONLY the marked, unreferenced declaration and keeps the tree
// building. After the unused-variable fix deletes unusedScratch, `compile` still
// passes and `detector-clears` reports zero matches.
func Scale(n int) int {
	return n * scaleFactor
}

File: species/fixture/testdata/unused-variable/repo/repo.ts
// sum returns a + b. The marked local `scratch` is computed and then never read.
// TypeScript does not treat an unused local as a compile error, so this is the
// meaningful in-function target for the indented delete-match path: the fix
// removes the indented `const` declaration with its leading whitespace intact,
// and detector-clears proves the declaration is gone (compile is a vacuous pass
// on this Go-module-only fixture, which has no TS toolchain wired).
export function sum(a: number, b: number): number {
  // ant:unused-variable scratch is computed but never read
  const scratch = a * b;
  return a + b;
}
